azurerm/resource_arm_private_link_service.go

```diff
@@ -0,0 +1,646 @@
+)
+
+func resourceArmPrivateLinkService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmPrivateLinkServiceCreateUpdate,
+		Read:   resourceArmPrivateLinkServiceRead,
+		Update: resourceArmPrivateLinkServiceCreateUpdate,
+		Delete: resourceArmPrivateLinkServiceDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
+
+			"auto_approval_subscription_ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validate.NoEmptyStrings,
+				},
+			},
+
+			"visibility_subscription_ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validate.NoEmptyStrings,
+				},
+			},
+
+			"fqdns": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validate.NoEmptyStrings,
+				},
+			},
+
+			"nat_ip_configuration": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.NoEmptyStrings,
+						},
+						"private_ip_allocation_method": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(network.Static),
+								string(network.Dynamic),
+							}, false),
+							Default: string(network.Dynamic),
+						},
+						"private_ip_address": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validate.NoEmptyStrings,
+						},
+						"private_ip_address_version": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(network.IPv4),
+								string(network.IPv6),
+							}, false),
+							Default: string(network.IPv4),
+						},
+						"primary": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  true,
+						},
+						"subnet_id": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.NoEmptyStrings,
+						},
+					},
+				},
+			},
+
+			"load_balancer_frontend_ip_configuration_ids": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validate.NoEmptyStrings,
+				},
+			},
+
+			"private_endpoint_connection": {
+				Type:     schema.TypeList,
+				Optional: true,
```

Comment:
@jeffreyCline , could you help to also add attribute "Computed: true,"? "make test" would compare this parameter between tfstate and tfconfig after created resource. So if this parameter isn't specified in tf config to create private link service and then associate private endpoint connection to existing private link service, you will find the count of this parameter in tfstate is "1", however it is "0" in tfconfig since we didn't specified this parameter in tfconfig so that "make test" would be failure. So we need to add both "Optional: true," and "Computed: true," for this parameter. Thanks for this.

It should be below this after updated.
"private_endpoint_connection": {
				Type:     schema.TypeList,
				Optional: true,
                                Computed: true,
                                Elem: &schema.Resource{

Reply:
Fixed.